File: scripts/test_client.py
import os, requests, json
BASE = os.getenv("GATEWAY_URL", "https://api.example.com")
KEY = os.getenv("GW_API_KEY", "devkey")
headers = {"x-api-key": KEY, "Content-Type": "application/json"}

print(requests.get(f"{BASE}/health/", headers=headers).json())

r = requests.post(f"{BASE}/rpc/proxy", headers=headers, json={"jsonrpc":"2.0","method":"eth_blockNumber","params":[],"id":1})
print(r.status_code, r.json())

File: scripts/test-client.js
#!/usr/bin/env node
import { GatewayClient } from "../sdk/ts-client/dist/index.js";

const BASE = process.env.GATEWAY_URL || "https://api.example.com";
const KEY = process.env.GW_API_KEY || "devkey";

async function run() {
  const c = new GatewayClient(BASE, KEY);
  console.log("Health:", await c.health());
  console.log("Block number:", await c.rpcProxy("eth_blockNumber"));
  // Submit dummy tx (non-signed); expect error or job id depending on server
  try {
    const res = await c.submitTx("evm:goerli", "0xDEADBEEF");
    console.log("Submit job:", res);
  } catch (e) {
    console.error("Submit failed:", e.message);
  }
}

run().catch(err => {
  console.error(err);
  process.exit(1);
});
